Sources/StreamChat/StateLayer/CurrentUser.swift

```diff
@@ -0,0 +1,105 @@
+//
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+/// An object which represents the currently logged in user.
+@available(iOS 13.0, *)
+public struct CurrentUser {
+    private let authenticationRepository: AuthenticationRepository
+    private let currentUserUpdater: CurrentUserUpdater
+    
+    init(user: CurrentChatUser, client: ChatClient, environment: Environment = .init()) {
+        authenticationRepository = client.authenticationRepository
+        currentUserUpdater = environment.currentUserUpdaterBuilder(
+            client.databaseContainer,
+            client.apiClient
+        )
+        state = environment.stateBuilder(
+            user,
+            client.databaseContainer
+        )
+    }
+    
+    /// An observable object representing the current state of the user.
+    public let state: CurrentUserState
+    
+    /// Updates the currently logged-in user data.
+    ///
+    /// - Note: Setting any arguments to nil will keep the existing value.
+    ///
+    /// - Parameters:
+    ///   - name: The name to be set to the user.
+    ///   - imageURL: The URL of the avatar image.
+    ///   - extraData: Additional data associated with the user.
+    ///
+    /// - Throws: An error while communicating with the Stream API or when user is not logged in.
+    public func updateUserData(name: String? = nil, imageURL: URL? = nil, extraData: [String: RawJSON] = [:]) async throws {
```

Comment:
The previous name is a bit redudant, maybe it is a good opportunity to change it?
```suggestion
    public func update(name: String? = nil, imageURL: URL? = nil, extraData: [String: RawJSON] = [:]) async throws {
```

Reply:
Great suggestion, also aligns with what `Chat` has for updating channel data.
```swift
// Chat.swift
public func update(
        name: String?,
        imageURL: URL?,
        team: String?,
        members: Set<UserId> = [],
        invites: Set<UserId> = [],
        extraData: [String: RawJSON] = [:]
    ) async throws {
```